pulp_rpm/app/models/repository.py

```diff
@@ -181,6 +184,28 @@ def _apply_retention_policy(self, new_version):
 
             new_version.remove_content(Content.objects.filter(pk__in=old_packages))
 
+    def _resolve_distribution_trees(self, new_version, previous_version):
+        """
+        There can be only one distribution tree in a repo version
+        :param new_version:
+        :param previous_version:
+        :return:
+        """
+        disttree_pulp_type = DistributionTree.get_pulp_type()
+        current_disttrees = DistributionTree.objects.filter(
+            pk__in=new_version.content.filter(pulp_type=disttree_pulp_type))
+        if len(current_disttrees) < 2:
+            return
+
+        if previous_version:
+            previous_disttree_pk = previous_version.content.get(pulp_type=disttree_pulp_type))
+            new_version.remove_content(Content.objects.filter(pk=previous_disttree_pk))
+            DistributionTree.objects.filter(pk=previous_disttree_pk).delete()
```

Comment:
Is it a correct way of removing a content unit completely?

Reply:
I don't get why we need to delete it,  only removing the content is not enough? Putting it better, if we find out the new DistributionTree is bad, it is possible to go back to the previous one?

Reply:
You are right. I keep wanting to get rid off kickstarts :)
The only time we need to delete them completely is when a repo is getting removed. Thanks